src/runtime/virtcontainers/kata_agent.go

```diff
@@ -281,23 +281,16 @@ type KataAgentState struct {
 
 // nolint: govet
 type kataAgent struct {
-	ctx      context.Context
-	vmSocket interface{}
-
-	client *kataclient.AgentClient
-
-	// lock protects the client pointer
-	sync.Mutex
-
-	state KataAgentState
-
+	ctx         context.Context
+	vmSocket    interface{}
+	client      *kataclient.AgentClient
 	reqHandlers map[string]reqFunc
+	state       KataAgentState
 	kmodules    []string
-
```

Comment:
This code cleanup also seems somewhat welcome, but belongs to a separate commit to help us figure out what is cleanup and what is actual logic change.

Reply:
Meh agree, let me remove that.